pkg/logs/decoder/decoder_test.go

```diff
@@ -170,3 +172,47 @@ func TestDecoderWithDockerHeader(t *testing.T) {
 	assert.Equal(t, expected, output.Content)
 	d.Stop()
 }
+
+func TestDecoderWithSingleline(t *testing.T) {
+	var output *Output
+
+	d := InitializeDecoder(config.NewLogSource("", &config.LogsConfig{}), parser.NoopParser)
+	d.Start()
+	defer d.Stop()
+
+	d.InputChan <- NewInput([]byte("message\n"))
+	output = <-d.OutputChan
+	assert.Equal(t, []byte("message"), output.Content)
```

Comment:
Why "\n" is removed here? where is this removal logic?

Reply:
https://github.com/DataDog/datadog-agent/pull/3655#discussion_r291734553